### PR TITLE
fix: resolve issue with placeholder & related code

### PR DESCRIPTION
**Pull request by pawanpaudel93**
## Summary

This PR resolves the issue with `generateLogoPlaceholder` and its usage. Related to https://github.com/arconnectio/ArConnect/pull/574

### CODE REVIEW

**Review by 7i7o — APPROVED** [review]
utACK 👍

**Review by 7i7o — APPROVED** [review]
tACK 👍 

We still need to fix the reset PR that clears some keys in storage that we need after a reset

File: src/utils/wallets/wallets.provider.tsx [hunk 1]
```diff
@@ -67,15 +67,15 @@ export function WalletsProvider({
 
       if (hasWallets && decryptionKey) {
```
Comment (Danziger) [inline src/utils/wallets/wallets.provider.tsx]: You can simplify this whole block:

```
if (hasWallets) {
  walletStatus = decryptionKey ? "unlocked" : "locked";
} else if (redirectToWelcome) { ... } 
```

File: src/utils/wallets/wallets.provider.tsx [hunk 2]
```diff
@@ -67,15 +67,15 @@ export function WalletsProvider({
...
-      } else if (!decryptionKey) {
+      } else if (!decryptionKey && hasWallets) {
         walletStatus = "locked";
       } else if (redirectToWelcome) {
         // This should only happen when opening the regular popup, but not for the auth popup, as the
         // `createAuthPopup` will open the welcome page directly, instead of the popup, if needed:
 
         openOrSelectWelcomePage(true);
 
-        window.top.close();
```
Comment (Danziger) [inline src/utils/wallets/wallets.provider.tsx:78]: This is needed, at least for the extension modal.

File: src/components/hardware/HardwareWalletTheme.tsx [hunk 3]
```diff
@@ -34,6 +34,14 @@ export function ArConnectThemeProvider({ children }: PropsWithChildren<{}>) {
   const hardwareApi = useHardwareApi();
   const theme = useTheme();
   const themeModifier = hardwareApi ? hardwareThemeModifier : noThemeModifier;
+  const isWelcomePage = window.location.pathname.includes("welcome");
+
+  useEffect(() => {
+    if (isWelcomePage) {
```
Comment (Danziger) [inline src/components/hardware/HardwareWalletTheme.tsx]: What was the issue with this? Just curious.

File: src/components/dashboard/Reset.tsx [hunk 4]
```diff
@@ -25,9 +25,14 @@ export function ResetDashboardView() {
...
   async function reset() {
     try {
-      // get all keys
+      // get all keys except gateways
       const allStoredKeys = Object.keys(
         (await browser.storage.local.get(null)) || {}
+      ).filter((key) => key !== "gateways");
+
+      // remove all keys except gateways
+      await Promise.allSettled(
```
Comment (Danziger) [inline src/components/dashboard/Reset.tsx]: This block is duplicated below (you'll see if if you expand the diff).

File: src/components/hardware/HardwareWalletTheme.tsx [hunk 5]
```diff
@@ -54,19 +56,27 @@ export function ArConnectThemeProvider({ children }: PropsWithChildren<{}>) {
...
+
+export function ThemeBackgroundObserver({
+  theme
+}: ThemeBackgroundObserverProps) {
   const styledComponentsTheme = useStyledComponentsTheme();
   const backgroundColor = styledComponentsTheme.background;
   const textColor = styledComponentsTheme.primaryText;
 
   useEffect(() => {
+    if (!theme) return;
```
Comment (Danziger) [inline src/components/hardware/HardwareWalletTheme.tsx:78]: `theme` becomes `undefined` when the extension popup closes, which was setting the wrong value in `localStorage` / CSS vars on the first render.

File: src/utils/storage.ts [hunk 6]
```diff
@@ -42,3 +43,25 @@ export interface RawStoredTransfer {
...
 }
+
+export const ARCONNECT_THEME_BACKGROUND_COLOR =
+  "ARCONNECT_THEME_BACKGROUND_COLOR";
+export const ARCONNECT_THEME_TEXT_COLOR = "ARCONNECT_THEME_TEXT_COLOR";
+
+/**
+ * Clear all storage keys except for gateways.
+ */
+export async function resetStorage() {
```
Comment (Danziger) [inline src/utils/storage.ts]: Moved the actual functionality here.

File: src/utils/wallets/wallets.provider.tsx [hunk 7]
```diff
@@ -65,17 +65,17 @@ export function WalletsProvider({
...
-        walletStatus = "unlocked";
-      } else if (!decryptionKey) {
-        walletStatus = "locked";
+      if (hasWallets) {
+        walletStatus = decryptionKey ? "unlocked" : "locked";
       } else if (redirectToWelcome) {
         // This should only happen when opening the regular popup, but not for the auth popup, as the
         // `createAuthPopup` will open the welcome page directly, instead of the popup, if needed:
 
         openOrSelectWelcomePage(true);
```
Comment (Danziger) [inline src/utils/wallets/wallets.provider.tsx]: @nicholaswma Any reason why an `await` here is necessary? When adding it, I see the extension popup appear for a fraction of a second, before the Welcome page opens on a new tab; without it, clicking the extension icon opens the Welcome page straight away without the popup flicker.

Comment (nicholaswma) [reply src/utils/wallets/wallets.provider.tsx]: @Danziger is this working without the await on your end? The welcome page never opens for me without it - I'm not sure if it's due to `window.top.close()` affecting it.

**Review by 7i7o — APPROVED** [review]
tACK 👍 

The reset fix is pending to be fixed in a separate PR

**Pull request merged**